### DISCUSSION TITLE
auth.updateUser not returning a response properly

### DISCUSSION THREAD

**Discussion by Manderson3568** [supabase/supabase#19011] [Questions] [answered]
Hi,
I am building a svelte kit project, this code exists inside a +page.svelte. This is the function I am currently using, and I have received similar results from a try/catch setup. 
I have been able to use all of the supabase auth session and login functions perfectly fine, as well as a full crud system on several tables, so I think my base setup is working fine. 
It also actually updates the password on the user account. 
From what I can tell is appears to get hung up in the await function, and leaves the page in a permanent limbo. None of the console logs trigger at all. 

```
    const handleReset = async () => {
        const { data: user, error } = await supabase.auth.updateUser({
            password: password,
            email: $profile.email
        });
        console.log(user);
        if (error) {
            console.log("error", error);
        } else if (data) {
            console.log("data", user);
        } else {
            console.log("Something");
        }
    };
```

    However When I go into chrome's network tab, the call appears successful, and the password updates. 
    
![image](https://github.com/supabase/supabase/assets/111272658/079a2e97-7c4b-4808-9a84-80e552e9d926)
![image](https://github.com/supabase/supabase/assets/111272658/3cb4a737-aa80-4bae-920e-a7c68e7d26f1)

If I trigger an error by putting in the current password I get more of a response.
![image](https://github.com/supabase/supabase/assets/111272658/1a0e1aae-5117-4960-a24b-0aa205f1505e)
![image](https://github.com/supabase/supabase/assets/111272658/f5a46178-3e0f-4c12-811b-59dcb3055bd8)
![image](https://github.com/supabase/supabase/assets/111272658/4e92664c-18bf-4164-98fd-f70be8ace9f3)

Any help would be appreciated.

**Comment by GaryAustin1** [comment]
Are you doing Supabase calls in the auth event handler by any chance?  There seems to be a bug if you are that causes the next Supabase auth call to fail.
https://github.com/supabase/gotrue-js/issues/762

**Reply by Manderson3568** [reply]
This is my first supabase project, I am only guessing what you are referring to. 
This is my hook.server.js

```
export const handle = async ({ event, resolve }) => {
    event.locals.supabase = createSupabaseServerClient({
        supabaseUrl: PUBLIC_SUPABASE_URL,
        supabaseKey: PUBLIC_SUPABASE_ANON_KEY,
        event
    });

    event.locals.getSession = async () => {
        const {
            data: { session }
        } = await event.locals.supabase.auth.getSession();
        return session;
    };
  return resolve(event, {
        filterSerializedResponseHeaders(name) {
            return name === "content-range";
        }
    });
};
```

and this is in my  root layout.svelte
 ```
 const {
            data: { subscription }
        } = supabase.auth.onAuthStateChange(async (event, _session) => {
            // console.log(_session);
            if (_session?.expires_at !== session?.expires_at) {
                invalidate("supabase:auth");
            }
            if (_session) {
                const { data, error } = await supabase
                    .from("profiles")
                    .select("*, business_users(admin,owner, id, businesses(id,name, financial))")
                    .eq("id", _session.user.id)
                    .single();
                if (!error) {
                    profile.set({ ...data });
                    console.log("logged in callback");
                } else {
                    console.log(error);
                }
            }
        });
```
If that doesn't answer the question could you clarify what you are referring to.

**Reply by GaryAustin1** [reply, marked as answer]
It does.  The link I gave shows that your onAuthStateChange handler is likely the cause of the issue because of a bug.  Towards the bottom of the link responses points out the issue and some things users did to fix it.   Doing a supabase call in the onAuthStateChange handler will cause at least the next auth call to hang even if in another part of your code.  So for now you can't make calls in that handler.

![image](https://github.com/supabase/supabase/assets/54564956/4f3b1259-62b0-4d51-8dfb-2f87c0405c37)
That code can't be there currently.

**Reply by Manderson3568** [reply]
Awesome thank you, I will take another look through that information now that I understand what you are referring to.

**Reply by Manderson3568** [reply]
You were 100% correct, moving that out of the callback sorted the problem, thanks again for the help.

**Comment by codebeast0420** [comment]
Hi @Manderson3568 
I'm facing same issue with you.
can you let me know how you fixed it?
